packages/bp/src/core/dialog/dialog-engine.ts

```diff
@@ -395,9 +409,7 @@ export class DialogEngine {
 
   protected async _transition(sessionId: string, event: IO.IncomingEvent, transitionTo: string) {
     let context: IO.DialogContext = event.state.context || {}
-    if (!event.activeProcessing?.errors?.length) {
```

Comment:
removing this condition is the most dangerous thing of this PR, but based on commit history, I believe this condition was only added to prevent the following scenario:

1. infinite loop is detected
2. error is caught and the dialog engine wants to transition to the error flow (with the same event stack trace)
3. infinite loop is again detected because stack trace has not changed

If my assumption is true, I can safely remove the condition as this PR specifically ensures that infinite loops are not handled by an error handling flow.